### PR TITLE
fix: make sure the memory subsystem is available before accessing the memory limit

### PR DESCRIPTION
**Pull request by kruskall**
## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

Closes https://github.com/elastic/apm-server/issues/10097

### CODE REVIEW

File: internal/beater/memlimit_cgroup.go [hunk 1]
```diff
@@ -55,6 +55,9 @@ func cgroupMemoryLimit(rdr *cgroup.Reader) (uint64, error) {
 		if err != nil {
 			return 0, fmt.Errorf("unable to read cgroup limits: %w", err)
 		}
+		if stats.Memory == nil {
+			return 0, fmt.Errorf("cgroup memory subsystem unavailable")
+		}
```
Comment (simitt) [inline internal/beater/memlimit_cgroup.go:60]: What if `stats.Memory` is available, but `stats.Memory.Mem.Max.Bytes` isn't? Would aligning the [behavior with the cgroupsV2 logic](https://github.com/elastic/apm-server/pull/10098/files#diff-9465d4addb3bf8eac10da4a31232f2e5911ed5840bcf9afb3f024f81730448b2R67) make sense?

Comment (kruskall) [reply internal/beater/memlimit_cgroup.go:60]: > What if stats.Memory is available, but stats.Memory.Mem.Max.Bytes isn't?

`stats.Memory.Mem.Limit.Bytes` is an `uint64` so the value would be 0

> Would aligning the [behavior with the cgroupsV2 logic](https://github.com/elastic/apm-server/pull/10098/files#diff-9465d4addb3bf8eac10da4a31232f2e5911ed5840bcf9afb3f024f81730448b2R67) make sense?

`stats.Memory.Mem.Max.Bytes` is an `opt.Uint` so the type is different. I don't think we can use the same method but the behaviour should be the same.

Comment (kruskall) [reply internal/beater/memlimit_cgroup.go:60]: To verify: 

```go
s := &cgroup.StatsV1{}
s.Memory = &cgv1.MemorySubsystem{}
fmt.Println(s.Memory.Mem.Limit.Bytes) # prints 0
```

**Review by simitt — APPROVED** [review]

**Review by simitt — APPROVED** [review]

**Pull request merged**